File: Taller1/Fibonacci.java
import matplotlib.pyplot as plt
from timeit import timeit
from time import time

def fibonacci(n):
    
    if n <= 1:
        return 1
    
    fib = 1
    prevFib = 1
    
    for i in range(2, n):
        temp = fib
        fib += prevFib
        prevFib = temp
    
    return fib
 
    
listaInputs = []
listaOutputs = []
listaTiempo = []
for i in range(1,5000):
    
    tiempo_inic = time()
    fibonacci(i)
    tiempo_fin = time()
    tiempo_ejec = (tiempo_fin - tiempo_inic)*1000000
        
    if tiempo_ejec < 20:
        listaTiempo.append(tiempo_ejec)
        listaOutputs.append(fibonacci(i))
        listaInputs.append(i)
       
    
print (listaInputs) 
print(listaOutputs)
print(listaTiempo)


x = listaInputs
y = listaTiempo

plt.scatter(x, y)


plt.xlabel('MSFT')
plt.ylabel('AAPL')
plt.title('Daily Prices in 2014');

File: Taller1/Fibonacci.c
#include <stdio.h>

unsigned int fibonacciInt(unsigned int n) {

    if (n <= 1) 
        return 1;

    unsigned int fib = 1;
    unsigned int prevFib = 1;

    for (unsigned int i = 2; i < n; i++) {

        unsigned int temp = fib;
        fib += prevFib;
        prevFib = temp;

    }

    return fib;

}

unsigned long int fibonacciLongInt(unsigned long int n) {

    if (n <= 1) 
        return 1;

    unsigned long int fib = 1;
    unsigned long int prevFib = 1;

    for (unsigned long int i = 2; i < n; i++) {

        unsigned long int temp = fib;
        fib += prevFib;
        prevFib = temp;

    }

    return fib;

}


unsigned long long int fibonacciLongLongInt(unsigned long long int n) {

    if (n <= 1) 
        return 1;

    unsigned long long int fib = 1;
    unsigned long long int prevFib = 1;

    for (unsigned long long int i = 2; i < n; i++) {

        unsigned long long int temp = fib;
        fib += prevFib;
        prevFib = temp;

    }

    return fib;

}

void main () {

    //unsigned int number;
    //unsigned int long number;
    //unsigned int long long number;


    //number = fibonacciInt(8);
    //number = fibonacciLongInt(8);

    //printf("%u\n", number);
    //printf("%lu\n", number);
    //printf("%llu\n", number);

    
}
